client/src/components/DynamicLink.tsx

```diff
@@ -0,0 +1,53 @@
+import { useCanEquip, useMyHash, useToInt } from "../hooks/useCall";
+import { $item, Placeholder } from "../util/makeValue";
+import MainLink from "./MainLink";
+
+interface Props {
+  linkedContent:
+    | Placeholder<"Item">
+    | Placeholder<"Familiar">
+    | Placeholder<"Skill">;
+}
+
+const DynamicLink: React.FC<Props> = ({ linkedContent }) => {
+  const myHash = useMyHash() ?? 0;
+  const isItem = linkedContent.objectType === "Item";
+  const isFamiliar = linkedContent.objectType === "Familiar";
+  const isSkill = linkedContent.objectType === "Skill";
+  const linkID = useToInt(linkedContent) ?? 1;
+  const isEquippable =
```

Comment:
this can be a normal && if the useCanEquip comes first. I'm not a huge fan of this pattern - it's a little too clever when we can just name intermediate variables to avoid conditional execution.